File: sdk/flutter/mics_client_sdk/test/ws_uri_builder_test.dart
import 'package:mics_client_sdk/src/ws_uri_builder.dart';
import 'package:test/test.dart';

void main() {
  test('buildWsUri should append required query params', () {
    final u = buildWsUri('ws://localhost:8080/ws', 't1', 'tok', 'dev1');
    expect(u.queryParameters['tenantId'], 't1');
    expect(u.queryParameters['token'], 'tok');
    expect(u.queryParameters['deviceId'], 'dev1');
  });

  test('buildWsUri should preserve existing query params', () {
    final u = buildWsUri('ws://localhost:8080/ws?x=1', 't1', 'tok', 'dev1');
    expect(u.queryParameters['x'], '1');
    expect(u.queryParameters['tenantId'], 't1');
  });
}

File: sdk/wechat/mics-client-sdk/src/wsUriBuilder.test.ts
import { describe, expect, it } from "vitest";

import { buildWsUrl } from "./wsUriBuilder";

describe("buildWsUrl", () => {
  it("appends required query params", () => {
    const url = buildWsUrl("ws://localhost:8080/ws", "t1", "tok", "dev1");
    expect(url).toContain("tenantId=t1");
    expect(url).toContain("token=tok");
    expect(url).toContain("deviceId=dev1");
  });

  it("preserves existing query params", () => {
    const url = buildWsUrl("ws://localhost:8080/ws?x=1", "t1", "tok", "dev1");
    expect(url).toContain("x=1");
  });
});

